src/AdoNetCore.AseClient/AseParameter.cs

```diff
@@ -1,44 +1,212 @@
+            SourceVersion = DataRowVersion.Default;
+            IsNullable = false;
+        }
+
+        /// <summary>
+        /// Constructor function for an <see cref="AseParameter" />instance.
+        /// </summary>
+        /// <param name="parameterName">The name of the parameter to add.</param>
+        /// <param name="value">An object that is the value of the parameter.</param>
+        public AseParameter(string parameterName, object value) : this()
+        {
+            Value = value ?? throw new ArgumentNullException();
+            ParameterName = parameterName;
+
+            if (value == DBNull.Value)
+            {
+                // Do nothing.
+            }
+            else if (value is int)
+            {
+                AseDbType = AseDbType.Integer;
+            }
+            else if (value is short)
+            {
+                AseDbType = AseDbType.SmallInt;
+            }
+            else if (value is long)
+            {
+                AseDbType = AseDbType.BigInt; // NOTE: - char is not supported by the SAP AseClient.
+            }
+            else if (value is uint)
+            {
+                AseDbType = AseDbType.UnsignedInteger; // NOTE: - uint is not supported by the SAP AseClient.
+            }
+            else if (value is ushort)
+            {
+                AseDbType = AseDbType.UnsignedSmallInt; // NOTE: - ushort is not supported by the SAP AseClient.
+            }
+            else if (value is ulong)
+            {
+                AseDbType = AseDbType.UnsignedBigInt; // NOTE: - ulong is not supported by the SAP AseClient.
+            }
+            else if (value is byte)
+            {
+                AseDbType = AseDbType.TinyInt;
+            }
+            else if (value is sbyte)
+            {
+                AseDbType = AseDbType.TinyInt;
+            }
+            else if (value is bool)
+            {
+                AseDbType = AseDbType.Bit;
+            }
+            else if (value is float)
+            {
+                AseDbType = AseDbType.Float;
+            }
+            else if (value is double)
+            {
+                AseDbType = AseDbType.Double;
+            }
+            else if (value is decimal)
+            {
+                AseDbType = AseDbType.Decimal;
+            }
+            else if (value is Guid)
+            {
+                AseDbType = AseDbType.Binary;
+                Size = 16;
+            }
+            else if (value is DateTime)
+            {
+                AseDbType = AseDbType.DateTime;
+            }
+            else if (value is byte[])
+            {
+                AseDbType = AseDbType.Binary;
+                Size = ((byte[])value).Length;
+            }
+            else if (value is char) // NOTE: - char is not supported by the SAP AseClient.
             {
-                return (DbType)_type;
+                AseDbType = AseDbType.UniChar;
+                Size = 1;
```

Comment:
`Size` should be the byte length, rather than the character length.
If we know the value, we can calculate it using `Encoding.GetByteCount(.)`. UniChar is UTF-16 encoded, so `Encoding.Unicode.GetByteCount(.)` would be appropriate.
If we don't know the value (such as for return/output parameters, the Size must be specified by the caller)
Note: if Size is 0 when we are generating tokens to send over the wire, the token generator should calculate it anyway.